### PR TITLE
Add test for issue #196

### PR DESCRIPTION
**Pull request by hubermat**
This PR adds a unit test for issue #196, proving that everything is all right.

It should be applied after PR #209 and PR #210, otherwise the unit tests will all fail.

### CODE REVIEW

File: tests/RecurrencesTest.php [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+        $ical = new ICal(false, $options);
+        $ical->initString($testIcal);
+
+        $events = $ical->events();
+
+        $this->assertCount($count, $events);
+
+        foreach($checks as $check) {
+            //echo $events[$check['index']]->dtstart_array[3].PHP_EOL;
+
```
Comment (u01jmg3) [inline tests/RecurrencesTest.php]: Can this be removed?

File: tests/RecurrencesTest.php [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+";
+    }
+
+    function getIcalFooter() {
+        return "END:VCALENDAR
+";
+    }
+
+
+}
```
Comment (u01jmg3) [inline tests/RecurrencesTest.php]: Rogue blank lines

File: tests/SingleEventsTest.php [hunk 3]
```diff
@@ -0,0 +1,452 @@
+<?php
+
+namespace Churchtools\Tests\ICal;
```
Comment (u01jmg3) [inline tests/SingleEventsTest.php]: Can this custom namespace be removed?

File: tests/SingleEventsTest.php [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+
+use ICal\ICal;
+use PHPUnit\Framework\TestCase;
+
+class SingleEventsTest extends TestCase {
+
+    public function testFullDayTimeZoneBerlin() {
+        $checks = [
+            ['index' => 0, 'dateString' => '20000301', 'message' => '1st event, CET: '],
+        ];
```
Comment (u01jmg3) [inline tests/SingleEventsTest.php]: Should this use the old array notation as the library supports PHP 5.3?

File: tests/icalmonthly.txt [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+UID:M2CD-1-1-5FB000FB-BBE4-4F3F-9E7E-217F1FF97208
+RRULE:FREQ=MONTHLY;BYMONTHDAY=1;WKST=SU;COUNT=25
+DTSTART;VALUE=DATE:20180701
+DTEND;VALUE=DATE:20180702
+SUMMARY:Monthly
+LAST-MODIFIED:20110429T222101Z
+DTSTAMP:20170630T105724Z
+SEQUENCE:0
+END:VEVENT
+END:VCALENDAR
```
Comment (u01jmg3) [inline tests/icalmonthly.txt]: Should this be called `icalmonthly.ics`?

File: tests/issue196.txt [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+SUMMARY:test
+RECURRENCE-ID;TZID=Europe/Berlin:20191110T190000
+DTSTART;TZID=Europe/Berlin:20191110T180000
+DTEND;TZID=Europe/Berlin:20191110T220000
+TRANSP:OPAQUE
+SEQUENCE:25
+X-MOZ-GENERATION:37
+DURATION:PT0S
+END:VEVENT
+END:VCALENDAR
```
Comment (u01jmg3) [inline tests/issue196.txt]: Should this be called `issue-196.ics`?

**Comment by u01jmg3** [comment]
ℹ️ Commenting on this PR as it is an accumulation of #209 and #210.

---

When I run `composer test` I am getting the following **errors** and **failures**:

```
There were 11 errors:

1) RecurrencesTest::testYearlyFullDayTimeZoneBerlin
PHPUnit_Framework_Exception:

Caused by exception 'ErrorException' with message 'unserialize(): Error at offset 0 of 4546 bytes' in C:\ics-parser\vendor\phpunit\phpunit\src\Util\PHP.php:114
```

And

```
There were 4 failures:

1) Churchtools\Tests\ICal\SingleEventsTest::testFullDayTimeZoneBerlin
Failed asserting that actual size 0 matches expected size 1.

C:\ics-parser\tests\SingleEventsTest.php:75
C:\ics-parser\tests\SingleEventsTest.php:20

2) Churchtools\Tests\ICal\SingleEventsTest::testSeveralFullDaysTimeZoneBerlin
Failed asserting that actual size 0 matches expected size 1.

C:\ics-parser\tests\SingleEventsTest.php:75
C:\ics-parser\tests\SingleEventsTest.php:32

3) Churchtools\Tests\ICal\SingleEventsTest::testEventTimeZoneUTC
Failed asserting that actual size 0 matches expected size 1.

C:\ics-parser\tests\SingleEventsTest.php:75
C:\ics-parser\tests\SingleEventsTest.php:45

4) Churchtools\Tests\ICal\SingleEventsTest::testEventTimeZoneBerlin
Failed asserting that actual size 0 matches expected size 1.
```

File: tests/RecurrencesTest.php (hunk 1 above)
Comment (hubermat) [reply tests/RecurrencesTest.php]: removed.

File: tests/RecurrencesTest.php (hunk 2 above)
Comment (hubermat) [reply tests/RecurrencesTest.php]: removed.

File: tests/SingleEventsTest.php (hunk 3 above)
Comment (hubermat) [reply tests/SingleEventsTest.php]: removed.

File: tests/SingleEventsTest.php (hunk 4 above)
Comment (hubermat) [reply tests/SingleEventsTest.php]: fixed.

File: tests/icalmonthly.txt (hunk 5 above)
Comment (hubermat) [reply tests/icalmonthly.txt]: renamed.

File: tests/issue196.txt (hunk 6 above)
Comment (hubermat) [reply tests/issue196.txt]: renamed.

**Comment by hubermat** [comment]
The errors and failures that you saw came from the annotation @runInSeparateProcess. This is a known issue under Windows. I simply removed the annotation, as there is no need to run the tests in separate processes. It runs even faster without! :-)

**Comment by u01jmg3** [comment]
Thanks for making all these fixes - I appreciate your time.

I am still not seeing green though?

```
There were 13 failures:

1) RecurrencesTest::testYearlyFullDayTimeZoneBerlin
Failed asserting that actual size 0 matches expected size 3.

C:\ics-parser\tests\RecurrencesTest.php:206
C:\ics-parser\tests\RecurrencesTest.php:32

2) RecurrencesTest::testMonthlyFullDayTimeZoneBerlin
Failed asserting that actual size 0 matches expected size 3.

C:\ics-parser\tests\RecurrencesTest.php:206
C:\ics-parser\tests\RecurrencesTest.php:47

3) RecurrencesTest::testMonthlyFullDayTimeZoneBerlinSummerTime
Failed asserting that actual size 0 matches expected size 3.

C:\ics-parser\tests\RecurrencesTest.php:206
C:\ics-parser\tests\RecurrencesTest.php:62

4) RecurrencesTest::testWeeklyFullDayTimeZoneBerlin
Failed asserting that actual size 0 matches expected size 6.

C:\ics-parser\tests\RecurrencesTest.php:206
C:\ics-parser\tests\RecurrencesTest.php:109

5) RecurrencesTest::testDailyFullDayTimeZoneBerlin
Failed asserting that actual size 0 matches expected size 31.

C:\ics-parser\tests\RecurrencesTest.php:206
C:\ics-parser\tests\RecurrencesTest.php:124

6) RecurrencesTest::testWeeklyFullDayTimeZoneBerlinLocal
Failed asserting that actual size 0 matches expected size 6.

C:\ics-parser\tests\RecurrencesTest.php:206
C:\ics-parser\tests\RecurrencesTest.php:142

7) RecurrencesTest::testRFCDaily10NewYork
Failed asserting that actual size 0 matches expected size 10.

C:\ics-parser\tests\RecurrencesTest.php:206
C:\ics-parser\tests\RecurrencesTest.php:158

8) RecurrencesTest::testRFCDaily10Berlin
Failed asserting that actual size 0 matches expected size 10.

C:\ics-parser\tests\RecurrencesTest.php:206
C:\ics-parser\tests\RecurrencesTest.php:174

9) RecurrencesTest::testRFCDaily10BerlinFromNewYork
Failed asserting that actual size 0 matches expected size 10.

C:\ics-parser\tests\RecurrencesTest.php:206
C:\ics-parser\tests\RecurrencesTest.php:190

10) SingleEventsTest::testFullDayTimeZoneBerlin
Failed asserting that actual size 0 matches expected size 1.

C:\ics-parser\tests\SingleEventsTest.php:83
C:\ics-parser\tests\SingleEventsTest.php:28

11) SingleEventsTest::testSeveralFullDaysTimeZoneBerlin
Failed asserting that actual size 0 matches expected size 1.

C:\ics-parser\tests\SingleEventsTest.php:83
C:\ics-parser\tests\SingleEventsTest.php:40

12) SingleEventsTest::testEventTimeZoneUTC
Failed asserting that actual size 0 matches expected size 1.

C:\ics-parser\tests\SingleEventsTest.php:83
C:\ics-parser\tests\SingleEventsTest.php:53

13) SingleEventsTest::testEventTimeZoneBerlin
Failed asserting that actual size 0 matches expected size 1.

C:\ics-parser\tests\SingleEventsTest.php:83
C:\ics-parser\tests\SingleEventsTest.php:66

FAILURES!
Tests: 15, Assertions: 33, Failures: 13.
Script phpunit --colors=always handling the test event returned with error code 1
```

**Comment by u01jmg3** [comment]
There were various issues related to line endings and `unfold()` not being able to expand the iCal strings in your tests. Will fix on my end to get these PRs merged.

**Pull request merged**